### DISCUSSION TITLE
Allow user to specify deadlines for habits

### DISCUSSION THREAD

**Discussion by ChristophyBarth** [iSoron/uhabits#493] [Feature Requests]
**Component**

The component I want to suggest is a deadline feature for the Loop Habit Tracker. 

---

**Description**

I so love this app that I promoted it in a heavy blog post yesterday, but then, I discovered that something was missing in the app. A deadline feature is a very very important one as this app helps people to achieve their long term goals. I am suggesting that a deadline feature should be added to the app. This feature will help users to set a deadline for every single goal he/she creates. Therefore, users will be more inspired and motivated to accomplish their goal before the deadline. 

---

**Mockup/Example**

Below is an example of the feature which I'm suggesting. It's a screenshot from an app. 
![20190512-132714](https://user-images.githubusercontent.com/49325171/57582520-aeb13200-74bd-11e9-8a92-11c59fa5f9c8.png)

---

**Benefit**

If this feature is implemented, it'll serve as a motivation/help increase the motivation of the user to keep up with his/her good habits; users will always endeavor to accomplish their goals.

---

**My Github**

 [Chrisbarth01](https://github.com/chrisbarth01)

**Comment by Vindexus** [comment]
This could be very cool. What would count as success? Would it be a certain score by a given date?

**Comment by ChristophyBarth** [comment]
Success would be when you perform that task in the number of days you had set...

**Comment by Vindexus** [comment]
Oh just the once? I guess was thinking it was more like "I have developed
this habit" as a goal, rather than "I have done it once" as a goal.

**Comment by iSoron** [comment]
Thank you for the suggestion, @chrisbarth01. A few other users have requested the same feature so I'm keeping this issue open for discussion, but honestly I still don't quite understand its purpose.

**Comment by vahidri** [comment]
@iSoron maybe same as a deadline for any goal? I mean a benchmark to hit, specially in the beginning of building a habit.

(by this rationale the strength (%) metric could be a/the criterion)

**Comment by ChristophyBarth** [comment]
> Thank you for the suggestion, @chrisbarth01. A few other users have requested the same feature so I'm keeping this issue open for discussion, but honestly I still don't quite understand its purpose.

How do you act when you know you've got a deadline to do something? You endeavour to make sure it's done.... right? That's what I'm suggesting. You could add the feature so it'll trigger users to work hard towards achieving their set goals.

**Comment by chennemann** [comment]
I think the question is more in what context exactly it would be beneficial. I don't quite understand the necessity for this feature either. 
If you want to track a habit for a given timeframe you can just delete it afterwards. I don't see why the app should further try to motivate or punish deviation from the expected repetitions. The strength metric was introduced to mitigate the Demotivation experienced after loosing the streak. When we would introduce deadlines I see this mitigation negated because you now need to compete with the deadline on a daily basis to keep on track (even though the score might still be ok) which might lead to Demotivation along the important progression of the habit. 

I hope I could make my point clear enough. What are your thoughts on this?
